### ISSUE TITLE
DHF 5.0.1: makeEnvelope() should accept a Sequence for headers

### ISSUE THREAD

**Issue by crism** [marklogic/marklogic-data-hub#2736]
### The issue

When `makeEnvelope()` is called with XML data format and a `Sequence` of `Element` instances as its second parameter, it errors out with `XDMP-ARGTYPE` when calling `xdmp.nodeKind()`.

I would expect the resulting envelope to have a `<headers>` element containing the contents of the `Sequence`. 

#### Tech details

Which Operating System are you using? CentOS 7.

Which version of MarkLogic are you using? 9.0-9.

Which version of the Data Hub Framework are you using? 5.0.1

Is this a QuickStart UI bug? No.

#### The devil is in the details...

```
Error running step: XDMP-ARGTYPE: xdmp.nodeKind(Sequence(Element(\"<agile-mastering:id xmlns:agile-mastering='http://marklogic.com/agile-mastering'/>\"), Element(\"<agile-mastering:dateTime xmlns:agile-mastering='http://marklogic.com/agile-mastering'>...</agile-mastering:dateTime>\"))) -- arg1 is not of type Node.
```

**Comment by crism** [comment]
Looking at [the way `headers` is used](https://github.com/marklogic/marklogic-data-hub/blob/master/marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow-utils.sjs#L144), it seems clear that a `Sequence` should be accepted here. However, the first think `makeEnvelope()` does is [to call `cleanData()`](https://github.com/marklogic/marklogic-data-hub/blob/master/marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow-utils.sjs#L86).

There appear to be logical errors [in `cleanData()`](https://github.com/marklogic/marklogic-data-hub/blob/master/marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow-utils.sjs#L217); if the format is XML, it calls `xdmp.nodeKind()`, resulting in this error, but it then asks if the payload is an `Array` and other questions that are inconsistent with `resp` being XML. This logic should be cleaned up, and should accept a `Sequence` of `Elements` as well.

**Comment by aebadirad** [comment]
Released as part of 5.0.2.
